src/main/java/edu/umd/lib/process/WufooProcessor.java

```diff
@@ -58,18 +58,34 @@ public WufooProcessor() {
   public void process(Exchange exchange) throws Exception {
 
     String message = exchange.getIn().getBody(String.class);
-    Map<String, List<String>> parameters = getQueryParams(message);
 
+    Map<String, List<String>> parameters = getQueryParams(message);
     checkHandshake(parameters);
+    String formName = getFormName(parameters.get("FormStructure"));
     Map<String, String> fields = getFields(parameters);
     JSONArray fieldsList = getFieldStructure(parameters.get("FieldStructure"), fields);
     HashMap<String, String> values = extractParameters(fieldsList);
+    values.put("FormName", formName);
 
     exchange.getOut().setBody(values);
     log.info("Total Number of Parameters from the request:" + parameters.size());
 
-    // SysAidConnector sysaid = new SysAidConnector();
-    // sysaid.createServiceRequest(values);
+  }
+
+  /****
+   * Get Form Name from the Wufoo Request
+   *
+   * @param formStructureArray
+   * @return
+   */
+  public String getFormName(List<String> formStructureArray) {
+    try {
+      JSONObject formStructure = new JSONObject(formStructureArray.get(0).toString());
+      return formStructure.getString("Name");
```

Comment:
We should not use this as the identifier for the mapping configuration. This value can contain spaces and other characters that could be problematic. Also, an immutable value (like form ID) should be preferred. If Wufoo does not provide a form ID, the handshake key or a hidden field can be used for this purpose.